src/buffer.cc

```diff
@@ -53,6 +54,41 @@ T* ValuesAs(uint8_t* values) {
   return reinterpret_cast<T*>(values);
 }
 
+template <typename T>
+float Sub(const uint8_t* buf1, const uint8_t* buf2) {
+  return static_cast<float>(
+    *reinterpret_cast<const T*>(buf1) - *reinterpret_cast<const T*>(buf2));
+}
+
+float CalculateDiff(
+    const Format::Component& comp, const uint8_t* buf1, const uint8_t* buf2) {
+  if (comp.IsInt8())
+    return Sub<int8_t>(buf1, buf2);
+  if (comp.IsInt16())
+    return Sub<int16_t>(buf1, buf2);
+  if (comp.IsInt32())
+    return Sub<int32_t>(buf1, buf2);
+  if (comp.IsInt64())
+    return Sub<int64_t>(buf1, buf2);
+  if (comp.IsUint8())
+    return Sub<uint8_t>(buf1, buf2);
+  if (comp.IsUint16())
+    return Sub<uint16_t>(buf1, buf2);
+  if (comp.IsUint32())
+    return Sub<uint32_t>(buf1, buf2);
+  if (comp.IsUint64())
+    return Sub<uint64_t>(buf1, buf2);
+  // TOOD(dsinclair): Handle float16 ...
+  if (comp.IsFloat16())
+    return 0.f;
```

Comment:
Should we assert false here? Or was this supposed to be implemented before merging?

Reply:
Should probably assert. I don't understand how float16 works at the moment so I'm skipping it for now. We have to verify all the float16 code in amber in general anyway.

Reply:
done